tracing-subscriber/src/layer.rs

```diff
@@ -808,6 +808,92 @@ where
     }
 }
 
+impl<L, S> Layer<S> for Option<L>
+where
+    L: Layer<S>,
+    S: Subscriber,
+{
+    #[inline]
+    fn new_span(&self, attrs: &span::Attributes<'_>, id: &span::Id, ctx: Context<'_, S>) {
+        if let Some(ref inner) = self {
+            inner.new_span(attrs, id, ctx)
+        }
+    }
+
+    #[inline]
+    fn register_callsite(&self, metadata: &'static Metadata<'static>) -> Interest {
+        match self {
+            Some(ref inner) => inner.register_callsite(metadata),
+            None => Interest::always(),
+        }
+    }
+
+    #[inline]
+    fn enabled(&self, metadata: &Metadata<'_>, ctx: Context<'_, S>) -> bool {
+        match self {
+            Some(ref inner) => inner.enabled(metadata, ctx),
+            None => true,
+        }
+    }
+
+    #[inline]
+    fn max_level_hint(&self) -> Option<LevelFilter> {
+        match self {
+            Some(ref inner) => inner.max_level_hint(),
+            None => None,
+        }
+    }
+
+    #[inline]
+    fn on_record(&self, _span: &span::Id, _values: &span::Record<'_>, _ctx: Context<'_, S>) {
+        if let Some(ref inner) = self {
+            inner.on_record(_span, _values, _ctx);
+        }
+    }
+
+    #[inline]
+    fn on_follows_from(&self, _span: &span::Id, _follows: &span::Id, _ctx: Context<'_, S>) {
+        if let Some(ref inner) = self {
+            inner.on_follows_from(_span, _follows, _ctx);
+        }
+    }
+
+    #[inline]
+    fn on_event(&self, _event: &Event<'_>, _ctx: Context<'_, S>) {
+        if let Some(ref inner) = self {
+            inner.on_event(_event, _ctx);
+        }
+    }
+
+    #[inline]
+    fn on_enter(&self, _id: &span::Id, _ctx: Context<'_, S>) {
+        if let Some(ref inner) = self {
+            inner.on_enter(_id, _ctx);
+        }
+    }
+
+    #[inline]
+    fn on_exit(&self, _id: &span::Id, _ctx: Context<'_, S>) {
+        if let Some(ref inner) = self {
+            inner.on_exit(_id, _ctx);
+        }
+    }
+
+    #[inline]
+    fn on_close(&self, _id: span::Id, _ctx: Context<'_, S>) {
+        if let Some(ref inner) = self {
+            inner.on_close(_id, _ctx);
+        }
+    }
+
+    #[inline]
```

Comment:
I added `#[inline]` to all the fn's as most of them seem small enough, and I could not find any online specific explanations on which fn's should be marked and which should not. So, Does that sound fine?

Reply:
I think that's correct.